skmob/measures/home_detection.py

```diff
@@ -0,0 +1,420 @@
+    for i, column in enumerate(dmat):
+        for j, distance in enumerate(dmat[column].values):
+            # since the matrix is symmetric, iterating on the upper matrix is enough for the computation 
+            if i == j:
+                break
+            
+            col_index = dmat.columns[i]
+            row_index = dmat.index[j]
+
+            # if the calculated distance is less than the specified radius, update the 'radius_activity' values
+            if distance <= radius:
+                col_activity = traj.at[col_index, constants.ACTIVITY]
+                row_activity = traj.at[row_index, constants.ACTIVITY]
+
+                traj.at[row_index, constants.RADIUS_ACTIVITY] += col_activity
+                traj.at[col_index, constants.RADIUS_ACTIVITY] += row_activity
+
+    # return the (latitude, longitude) tuple having the higher 'radius_activity' values
+    lat, lng = traj.sort_values(by=constants.RADIUS_ACTIVITY, ascending=False).index[0].split(' ')   
+    return (float(lat), float(lng))
+
+
+def _sjoin_activity(traj, radius):
+    # generate the index for the new GeoDataFrame
+    index = list(range(0,len(traj.index)))
+
+    # create a new GeoDataFrame, filling the 'geometry' attribute with the (longitude, latitude) TrajDataFrame points
+    # transform the geometries into an 'universal' crs, in order to treat the points in a two-dimensional space.
+    towers = gpd.GeoDataFrame(traj, index=index, geometry=gpd.points_from_xy(traj[constants.LONGITUDE], traj[constants.LATITUDE]),crs='EPSG:4326').to_crs(constants.UNIVERSAL_CRS)
+    towers['index'] = index
+    towers = towers.set_index('index')
+
+    # create a circular polygons, with a specified radius in kilometers, around the points arranged in the
+    # previous GeoDataFrame
+    circles = towers[['geometry']].copy()
+    circles['geometry'] = circles.buffer(radius * 1000)    
+
+    # perform a spatial join between all the polygons and the original points
+    # for each circle, 'radius_activity' values are calculated through the sum of the individual activities
+    points_within = towers.sjoin(circles, predicate='within').groupby('index_right')[[constants.ACTIVITY]].sum()
+    points_within.index.rename('index', inplace=True)
+    points_within.rename(columns={constants.ACTIVITY: constants.RADIUS_ACTIVITY}, inplace=True)
+    towers = towers.join(points_within)
+    towers = towers[[constants.LATITUDE, constants.LONGITUDE, constants.RADIUS_ACTIVITY]]    
+
+    # sort and return the (latitude, longitude) tuple having the higher 'radius_activity'
+    df = towers.rename(columns={0: constants.LATITUDE, 1: constants.LONGITUDE, 2: constants.RADIUS_ACTIVITY}).sort_values(by=constants.RADIUS_ACTIVITY, ascending=False).reset_index(drop=True)
+
+    lat = df.at[0, constants.LATITUDE]
+    lng = df.at[0, constants.LONGITUDE]
+
+    return (lat, lng)
+
+
+def _home_location_ma(traj, week_period, radius, mode): 
```

Comment:
why do you have a private and a public method for each alg? Can't you just have everything in the same method?